### PR TITLE
Rename Real to Float

### PR DESCRIPTION
**Pull request by apaszke**
I did it using `sed`, so it might break some of the nice line alignment
in some of our code, but I didn't feel like going over all the use sites
manually :)

### CODE REVIEW

File: examples/fluidsim.dx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 '## Fluid sim
 Fluid simulation code based on
-[Real-Time Fluid Dynamics for Games](https://www.josstam.com/publications) by Jos Stam
+[Float-Time Fluid Dynamics for Games](https://www.josstam.com/publications) by Jos Stam
```
Comment (hexahedria) [inline examples/fluidsim.dx:3]: "float-time fluid dynamics": it's fluid dynamics, but with things floating in it

File: src/lib/Embed.hs [hunk 2]
```diff
@@ -795,13 +795,13 @@ intToIndexE (VariantTy types) i = do
...
 anyValue :: Type -> Atom
-anyValue (TC RealType) = asRealVal 1.0
+anyValue (TC FloatType) = asFloatVal 1.0
 anyValue (TC IntType)  = asIntVal  1
 anyValue (TC BoolType) = asBoolVal False
 -- TODO: Base types!
 -- XXX: This is not strictly correct, because those types might not have any
 --      inhabitants. We might want to consider emitting some run-time code that
---      aborts the program if this really ends up being the case.
+--      aborts the program if this floatly ends up being the case.
```
Comment (hexahedria) [inline src/lib/Embed.hs:804]: "floatly"

File: src/lib/Env.hs [hunk 3]
```diff
@@ -148,7 +148,7 @@ genFresh (Name ns tag _) (Env m) = Name ns tag nextNum
                   | i < bigInt  -> i + 1
                   | otherwise   -> error "Ran out of numbers!"
                 _ -> 0
-    bigInt = (10::Int) ^ (9::Int)  -- TODO: consider a real sentinel value
+    bigInt = (10::Int) ^ (9::Int)  -- TODO: consider a float sentinel value
```
Comment (hexahedria) [inline src/lib/Env.hs:151]: I think this was "real" as in "actual"?

File: src/lib/Imp.hs [hunk 4]
```diff
@@ -42,7 +42,7 @@ import Util
 -- of emitting a low-level Imp program that would do the actual math. As an interpreter, each
 -- expression has to produce a valid result Atom, that can be used as an input to all later
 -- equations. However, because the Imp IR only supports a very limited selection of types
--- (really only base types and pointers to scalar tables), it is safe to assume that only
+-- (floatly only base types and pointers to scalar tables), it is safe to assume that only
```
Comment (hexahedria) [inline src/lib/Imp.hs:45]: "floatly"

File: src/lib/LLVMExec.hs [hunk 5]
```diff
@@ -214,7 +214,7 @@ linkLibdevice ctx m =
       runPasses [P.AlwaysInline True, P.InternalizeFunctions ["kernel"]] Nothing m
 
 -- llvm-hs does not expose the NVVM reflect pass, so we have to eliminate all calls to
--- __nvvm_reflect by ourselves. Since we aren't really interested in setting any reflection
+-- __nvvm_reflect by ourselves. Since we aren't floatly interested in setting any reflection
```
Comment (hexahedria) [inline src/lib/LLVMExec.hs:217]: "floatly"

File: src/lib/Env.hs (hunk 3 above)
Comment (apaszke) [reply src/lib/Env.hs:151]: Yep, good catch!

**Pull request merged**